backend/controllers/inviteContacts.ts

```diff
@@ -35,4 +36,57 @@ export default class InviteContactController {
       })
     }
   }
+
+  static async verifyPlanInvite(req: any, res: Response, next: NextFunction): Promise<void> {
+    try {
+      const planInviteID = req.params.id
+
+      // if no cookie, redirect to log in
+      if (req.user === null || req.user === undefined) {
+        res.status(200).json({
+          reason: 'NOT_LOGGED_IN',
+          planInviteID
+        })
+        return
+      }
+
+      const isEmailMatch = await PlanInviteService.isUserEmailMatchPlanInviteEmail(req?.user?.email, planInviteID)
+
+      if (isEmailMatch === null) throw new Error('fail to run isUserEmailMatchPlanInviteEmail')
```

Comment:
Do we need this error? If the email check doesnt run, and returns null is that any different than just email and logged in acc not matching?

Reply:
That's very true..if `isEmailMatch` is null then the user's email did not match too